src/detect-template-buffer.c

```diff
@@ -48,6 +48,9 @@ static int DetectTemplateBufferSetup(DetectEngineCtx *de_ctx, Signature *s,
 {
     s->list = DETECT_SM_LIST_TEMPLATE_BUFFER_MATCH;
     s->alproto = ALPROTO_TEMPLATE;
+    s->flags |= SIG_FLAG_STATE_MATCH;
```

Comment:
One way would be to make it part of the register function, but I guess there are some corner cases to consider. There are some keywords that depending on how and where they are used (e.g. lua) should have this flag or not.